src/core/src/core_logic/PatchInstaller.py

```diff
@@ -282,7 +349,202 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
+                    skip_packages.append(packages[index])
+                    self.status_handler.set_package_install_status(package_manager.get_product_name(packages[index]), str(package_versions[index]), Constants.NOT_SELECTED)
+                elif packages[index] not in self.last_still_needed_packages:
+                    # Could have got installed as dependent package of some other package. Package installation status could also have been set.
+                    already_installed_packages.append(packages[index])
+                    self.attempted_parent_package_install_count += 1
+                    self.successful_parent_package_install_count += 1
+                else:
+                    packages_in_batch.append(packages[index])
+                    package_versions_in_batch.append(package_versions[index])
+
+            if len(already_installed_packages) > 0:
+                self.composite_logger.log("Following packages are already installed. Could have got installed as dependent package of some other package " + str(already_installed_packages))
+
+            if len(skip_packages) > 0:
+                self.composite_logger.log("[Skipping packages " + str(skip_packages) + " - requires Ubuntu Advantage for Infrastructure with Extended Security Maintenance]")
+
+            if len(packages_in_batch) == 0:
+                continue
+
+            remaining_time = maintenance_window.get_remaining_time_in_minutes()
+
+            if maintenance_window.is_package_install_time_available(remaining_time, len(packages_in_batch)) is False:
+                self.composite_logger.log("Stopped installing packages in batches as it is past the maintenance window cutoff time for installing in batches." +
+                                           " Batch Index: {0}, remaining time: {1}, number of packages in batch: {2}".format(batch_index, remaining_time, str(len(packages_in_batch))))
+                maintenance_window_batch_cutoff_reached = True
+                remaining_packages = packages[begin_index:]
+                remaining_package_versions = package_versions[begin_index:]
+                break
+
+            # point in time status
+            progress_status = self.progress_template.format(str(datetime.timedelta(minutes=remaining_time)), str(self.attempted_parent_package_install_count), str(self.successful_parent_package_install_count), str(self.failed_parent_package_install_count), str(installed_update_count - self.successful_parent_package_install_count),
+                                                            "Processing batch index: " + str(batch_index) + ", Number of packages: " + str(len(packages_in_batch)) + "\nProcessing packages: " + str(packages_in_batch))
+            self.composite_logger.log(progress_status)
+
+            # package_and_dependencies initially conains only packages in batch. The dependencies are added in the list by method include_dependencies
```

Comment:
@rane-rajasi - note as reviewer - there is opportunity for code de-duplication. Came up as a flag since this typo was repeated.